src/_pytest/assertion/rewrite.py

```diff
@@ -459,39 +463,40 @@ def _fix(node, lineno, col_offset):
     return node
 
 
-def _get_assertion_exprs(src: bytes):  # -> Dict[int, str]
+def _get_assertion_exprs(src: bytes) -> Dict[int, str]:
     """Returns a mapping from {lineno: "assertion test expression"}"""
-    ret = {}
+    ret: Dict[int, str] = {}
 
     depth = 0
-    lines = []
-    assert_lineno = None
-    seen_lines = set()
+    lines: List[str] = []
+    assert_lineno: Optional[int] = None
+    seen_lines: Set[int] = set()
 
     def _write_and_reset() -> None:
         nonlocal depth, lines, assert_lineno, seen_lines
+        assert assert_lineno is not None
         ret[assert_lineno] = "".join(lines).rstrip().rstrip("\\")
         depth = 0
         lines = []
         assert_lineno = None
         seen_lines = set()
 
     tokens = tokenize.tokenize(io.BytesIO(src).readline)
-    for tp, src, (lineno, offset), _, line in tokens:
-        if tp == tokenize.NAME and src == "assert":
+    for tp, string, (lineno, offset), _, line in tokens:
```

Comment:
`string` shadows the module by the same name -- perhaps choose a different name here? -- smaller diff would be to rename the parameter to `source` instead